tensornetwork/tn_keras/entangler.py

```diff
@@ -102,24 +102,32 @@ def is_perfect_root(n, n_nodes):
 
     super(DenseEntangler, self).build(input_shape)
 
-    # Ensure input dim and output dim match
-    assert (
-        input_shape[-1] == self.output_dim
-    ), f'Input dim {input_shape[-1]} not equal to output dim {self.output_dim}'
-
     # Ensure the Entangler dimensions will work
     assert (
         is_perfect_root(input_shape[-1], self.num_legs)
     ), f'Input dim {input_shape[-1]}**(1. / {self.num_legs}) must be round.'
 
+    assert (
+        is_perfect_root(self.output_dim, self.num_legs)
+    ), f'Output dim {self.output_dim}**(1. / {self.num_legs}) must be round.'
+
     self.leg_dim = round(input_shape[-1]**(1. / self.num_legs))
-    self.num_nodes = self.num_levels
+    self.out_leg_dim = round(self.output_dim**(1. / self.num_legs))
+    self.num_nodes = self.num_levels * (self.num_legs - 1)
```

Comment:
Can you add a note to the docstring that makes it clear `num_levels` represents *complete* levels i.e. more than 1 node per level?